crates/iota/src/validator_commands.rs

```diff
@@ -1052,23 +1053,19 @@ pub async fn get_validator_summary(
     };
 
     // Check inactive
-    let mut cursor = None;
-    let mut has_next_page = true;
-    while has_next_page {
-        let page = client
+    let mut stream = GetAllPages::get_all_pages_stream(async |cursor| {
+        client
             .read_api()
             .get_dynamic_fields(inactive_pools_id, cursor, None)
-            .await?;
-        for dynamic_field_info in page.data {
-            let validator_summary =
-                get_validator_summary_from_validator_wrapper(client, dynamic_field_info.object_id)
-                    .await?;
-            if validator_summary.iota_address == validator_address {
-                return Ok(Some((ValidatorStatus::Inactive, validator_summary)));
-            }
+            .await
+    });
+    while let Some(dynamic_field_info) = stream.try_next().await? {
```

Comment:
Maybe a dumb question but could we not provide a `try_next` method ourselves so people don't have to import both `GetAllPages` and `TryStreamExt` ?